### ISSUE TITLE
[Enhancement] custom logits_processings

### ISSUE THREAD

**Issue by atawari** [EvolvingLMMs-Lab/lmms-eval#181]
Is there any plan to add custom logits processing
during evaluation like: https://github.com/huggingface/transformers/pull/12219#issue-923570044

**Comment by kcz358** [comment]
Hi, I don't think there is such option to do so currently. If you want to do so, I think you have to manually change the logits processing in the generate until function for the model you want to change.

**Comment by atawari** [comment]
Thanks. If not the custom logit processor, it would be nice to support all the `gen_kwargs` supported by the HF model. 
E.g. https://github.com/EvolvingLMMs-Lab/lmms-eval/blob/main/lmms_eval/models/llava_hf.py#L363

```
cont = self.model.generate(**inputs, **gen_kwargs, use_cache=self.use_cache)
```

user can specify the appropriate `gen_kwargs`.

**Comment by lynshwoo2022** [comment]
where can we change gen_kwargs for different benchmark?
